custom_components/zwave_mqtt/entity.py

```diff
@@ -135,19 +142,36 @@ def __init__(self, values):
         self.values = values
         self.options = values._options
         self.values._entity = self
+        self._dispatcher_callbacks = []
 
     @callback
     def value_changed(self, value):
         """Call when the value is changed."""
         if value.value_id_key in (v.value_id_key for v in self.values if v):
             self.async_schedule_update_ha_state()
 
+    @callback
     def value_added(self):
         """Handle a new value for this entity."""
 
+    @callback
+    def instance_updated(self, new_status):
+        """Call when the instance status changes."""
+        self.async_schedule_update_ha_state()
+
     async def async_added_to_hass(self):
         """Call when entity is added."""
         self.options.listen(EVENT_VALUE_CHANGED, self.value_changed)
+        self.options.listen(EVENT_INSTANCE_STATUS_CHANGED, self.instance_updated)
+        # add dispatcher callbacks in a list so we can clean them up again later
+        self._dispatcher_callbacks = [
```

Comment:
We can use `self.async_on_remove` to collect callbacks that should be called on entity remove.

Reply:
@MartinHjelmare isn't that where async_will_remove_from_hass is for ?

Reply:
ah I see what you mean now. It already has the functionality to execute the cleanup

Reply:
OK, changed. Can you take another look @MartinHjelmare